LuLu/Extension/LuLuServer.m

```diff
@@ -0,0 +1,180 @@
+//
+//  file: LuLuServer.m
+//  project: LuLu (launch daemon)
+//  description: handles remote calls to LuLu Server
+//
+//
+
+#import "consts.h"
+
+#import "Rule.h"
+#import "Rules.h"
+#import "Process.h"
+#import "utilities.h"
+#import "Preferences.h"
+#import "LuLuServer.h"
+
+//log handle
+extern os_log_t logHandle;
+
+@implementation LuLuServer
+
+-(NSDictionary*)getDefaultRules {
+    
+    //defining the url to make the request
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@/default-user-rules", LULU_SERVER_URL]];
+    
+    //blocking the space for the response
+    __block NSDictionary *result = nil;
+
+    //request
+    NSMutableURLRequest *request = nil;
+    
+    //wait semaphore
+    dispatch_semaphore_t semaphore = 0;
+
+    //init wait semaphore
+    semaphore = dispatch_semaphore_create(0);
+    
+    //alloc/init request
+    request = [[NSMutableURLRequest alloc] initWithURL:url];
+    
+    //set method type
+    [request setHTTPMethod:@"GET"];
+
+    [[[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData* data, NSURLResponse* response, NSError* error)
+    {
+        os_log_debug(logHandle, "got response %lu", (long)((NSHTTPURLResponse *)response).statusCode);
+        //sanity check(s)
+        if( (nil != data) &&
+            (nil == error) &&
+            (200 == (long)((NSHTTPURLResponse *)response).statusCode))
+        {
+            //serialize response into NSData obj
+            // wrap since we are serializing JSON
+            @try
+            {
+                //serialized
+                result = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
+                
+            }
+            //error converting
+            @catch (NSException *exception)
+            {
+                //err msg
+                os_log_error(logHandle, "ERROR: converting response %{public}@ to JSON threw %{public}@", data, exception);
+            }
+        }
+        //error making request
+        else
+        {
+            //err msg
+            os_log_error(logHandle, "ERROR: failed to get default rules from LuLuServer (%{public}@, %{public}@)", error, response);
+        }
+        
+        //trigger
+        dispatch_semaphore_signal(semaphore);
+        
+    }] resume];
+    
+    //wait for request to complete
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    
+
+    return result;
+}
+
+- (void) logNewAccessWithFlowUUID:(NSString*)flowUUID processName:(NSString*)processName {
```

Comment:
Adicionar documentação. Pelo nome do método eu não entendi bem oq ele faz. Acho que da para agnt renomear aqui tambem. "NewAccess" representa um flow ? Então podemos continuar chamando de Flow + a ação que ele faz para o servidor.

Reply:
Vi aqui pelo metodo que chama esse que a ideia dele é informar o servidor quando ocorre um not-matched access. O nome do método tem que representar essa ação.